src/clib/pio_darray.c

```diff
@@ -1431,7 +1431,7 @@ static int check_adios2_need_to_flush(file_desc_t *file, adios_var_desc_t *av)
     } \
 }
 
-#define ADIOS_CONVERT_FROM(FROM_TYPE_ID, from_type) \
+#define ADIOS_CONVERT_FROM(FROM_TYPE_ID, from_type, array, arraylen, buf) \
```

Comment:
I think this change should be enough for this PR. However there are some changes that might be good to introduce in a future PR,
# ADIOS_CONVERT_ARRAY could use from_buf, to_buf instead of array etc as args
# ADIOS_CONVERT_FROM macro should not be silently ignoring the out of memory error
# PIOc_convert_buffer_adios() could return an error code instead of the new buffer (and accept the new buffer ppointer as arg)
# You might also want to consider if memory allocation outside the conversion macro (instead of inside it) is a possibility (then PIOc_convert_buffer_adios() could just accept the new buffer as an argument & you just need to do sanity checks inside these macros/funcs)